swap_meet/vendor.py

```diff
@@ -1,2 +1,82 @@
+from swap_meet.item_helper_functions import get_highest_item
+from swap_meet.item_helper_functions import get_newest_item
+
+
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self, inventory=None):
+        self.inventory = [] if inventory is None else inventory
+
+
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+
+
+    def remove(self, item):
+        if item not in self.inventory:
+            return False
+
+        self.inventory.remove(item)
+        return item
+
+
+    def get_by_id(self, item_id):
+        for item in self.inventory:
+            if item.id == item_id:
+                return item
+        
+
+    def swap_items(self, other_vendor, my_item, their_item):
+        if my_item not in self.inventory or their_item not in other_vendor.inventory:
+            return False
+        
+        my_item_index = self.inventory.index(my_item)
+        their_item_index = other_vendor.inventory.index(their_item)
+        other_vendor.inventory[their_item_index], self.inventory[my_item_index] = my_item, their_item
+        return True
+    
+
+    def swap_first_item(self, other_vendor):
+        if not self.inventory or not other_vendor.inventory:
+            return False
+        
+        first_item = self.inventory[0]
+        first_item_other = other_vendor.inventory[0]
+
+        if  first_item  and first_item_other:
+            my_item_index = self.inventory.index(first_item)
+            their_item_index = other_vendor.inventory.index(first_item_other)
+            other_vendor.inventory[their_item_index], self.inventory[my_item_index] = first_item, first_item_other
+            return True
```

Comment:
Nit on line 46 about white spaces:
```suggestion
        if first_item and first_item_other:
            my_item_index = self.inventory.index(first_item)
            their_item_index = other_vendor.inventory.index(first_item_other)
            other_vendor.inventory[their_item_index], self.inventory[my_item_index] = first_item, first_item_other
            return True
```

Also, do we need to write the logic on lines 47-50? Can't we just invoke `swap_items`?

```py
def swap_first_item(self, other_vendor):
        if not self.inventory or not other_vendor.inventory:
            return False

        first_item = self.inventory[0]
        first_item_other = other_vendor.inventory[0]

        if  first_item  and first_item_other:
            return swap_items(other_vendor, first_item, first_item_other)
```